### ISSUE TITLE
Find the Longest Word in a String

### ISSUE THREAD

**Issue by CerealNopon** [freeCodeCamp/freeCodeCamp#50342]
### Describe the Issue

Test accepts two different inputs as correct answers when it shouldn't be. 

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-algorithm-scripting/find-the-longest-word-in-a-string

### Your code

```
function findLongestWordLength(str) {
  //arr1 is made using regexes
  //arr2 is made using array.split()
  
  let regex = /\w+/g;
  let arr1 = str.split(' ');
  let arr2 = str.match(regex);

  let maxLength1 = 0;
  let maxLength2 = 0;

  for (let i = 0; i < arr1.length; i++) {
    if (arr1[i].length > maxLength1) {
      maxLength1 = arr1[i].length;
    }
  }
  for (let j = 0; j < arr2.length; j++) {
    if (arr2[j].length > maxLength2) {
      maxLength2 = arr2[j].length;
    }
  }

  return maxLength2; //maxlength1 returns 11 while
                     //maxLength2 returns 6
                     //however, both pass the test correctly
}

console.log(findLongestWordLength("The quick-brown fox-jumped over the lazy dog"));
```

### Expected behavior

It should only have one correct integer response. Perhaps specify the inclusion of hyphens if they exist as one word or two words. 

### System

- Device: Laptop
- OS: Windows 10
- Browser: Chrome
- Version: ?

**Comment by TylerBettilyon** [comment]
I just checked this in the IDE for the problem, and I noticed a difference! In your console log, you're putting hyphens in between "quick-brown" and "fox-jumped", which is changing the number shown on the log. In the example, there are no hyphens: "The quick brown fox jumped over the lazy dog". Without the hyphens included, both of those solutions provide the same integer, and each of the test cases work correctly for both solutions you've made! The only hyphen present is in the last example, and it doesn't affect the test case since it's still shorter than the longest word even with the hyphen, so I think that this case is working as intended, unless FCC wants to adjust the testing criteria to be more strict :)
